### ISSUE TITLE
ToDo: diffs FF62-FF63

### ISSUE THREAD

**Issue by earthlng** [ghacksuserjs/ghacks-user.js#501]
FF63 is [scheduled](https://wiki.mozilla.org/Release_Management/Calendar) for release 23th Oct

[FF63 Release notes](https://www.mozilla.org/en-US/firefox/63.0/releasenotes/) [when ready]
[FF63 for developers](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/63)
[FF63 compatibility](https://www.fxsitecompat.com/en-CA/versions/63/)
[FF63 security advisories](https://www.mozilla.org/en-US/security/advisories/mfsa2018-26/)

199 diffs ( 122 new, 56 gone, 21 different )

### misc:
* [x] deal with Activity-Stream since it's no longer a system addon in FF63 - https://github.com/ghacksuserjs/ghacks-user.js/commit/58fa4e9b6d44ba37adf2d5083b09f28ef792dc06

### new in v63.0:
* [x] pref("browser.contentblocking.enabled", true); - [1486092,1482281,1476217](https://bugzilla.mozilla.org/buglist.cgi?bug_id=1486092,1482281,1476217) - https://github.com/ghacksuserjs/ghacks-user.js/commit/8fd6061bcc977d6e242ba08b1c50ac46b0225b7d
* [x] pref("media.autoplay.default", 0); - https://github.com/ghacksuserjs/ghacks-user.js/commit/56206f77ba6833876931e346212e3df9d2cf9ae5#diff-b00df6cb4137c00991376cd319d35348R1036
  * we'll deal with the rest of `media.autoplay.*` in FF64
* [x] pref("media.media-capabilities.enabled", true); - [1480190,1409664](https://bugzilla.mozilla.org/buglist.cgi?bug_id=1480190,1409664) - https://github.com/ghacksuserjs/ghacks-user.js/commit/f214e4bc4e69af62a031323142b3645bd0bad340 & https://github.com/ghacksuserjs/ghacks-user.js/commit/3423d39fa9255a4793fa60d6de63ba2cb2d95e1d
  * pref("media.media-capabilities.screen.enabled", false); - [1482243](https://bugzilla.mozilla.org/1482243)
* for visibility for those who want this:
  * pref("devtools.aboutdebugging.showSystemAddons", false);

### removed, renamed or hidden in v63.0:

- `ALL DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/56206f77ba6833876931e346212e3df9d2cf9ae5
* [x] `0202` - `browser.search.countryCode` (hidden pref) - [1462015](https://bugzilla.mozilla.org/1462015)
* [x] `0301a` - pref("app.update.enabled", true); - [1420514](https://bugzilla.mozilla.org/show_bug.cgi?id=1420514)
   * [x] update [wiki implementation page](https://github.com/ghacksuserjs/ghacks-user.js/wiki/1.3-Implementation)
* [x] `0402` - pref("services.blocklist.update_enabled", true); - [1458917](https://bugzilla.mozilla.org/show_bug.cgi?id=1458917)
* [x] `0503` - pref("shield.savant.enabled", false); - [1457226](https://bugzilla.mozilla.org/show_bug.cgi?id=1457226)
* [x] `0513` - Follow On Search is no longer a system addon, so we removed this item
* [x] `1031` - pref("browser.chrome.favicons", true); - [1453751](https://bugzilla.mozilla.org/show_bug.cgi?id=1453751) - merged into `browser.chrome.site_icons`
  * [x] fix the description: for a while now "bookmark favicons are stored in favicons.sqlite"
* [x] `2030` - pref("media.autoplay.enabled", true); - [1470082](https://bugzilla.mozilla.org/show_bug.cgi?id=1470082) - `replaced by media.autoplay.default`
* [x] `2704` - pref("network.cookie.lifetime.days", 90); - [1457170](https://bugzilla.mozilla.org/show_bug.cgi?id=1457170)
* [x] `5000's` - pref("browser.ctrlTab.previews", false); - [1473595](https://bugzilla.mozilla.org/show_bug.cgi?id=1473595)
- done earlier
* [x] pref("privacy.trackingprotection.ui.enabled", true); - [1476879](https://bugzilla.mozilla.org/1476879) - https://github.com/ghacksuserjs/ghacks-user.js/commit/b2fc9bc2664820cb22a658fb9a3977390029333a
  * note: REMOVED from the user.js as the default is true since FF57

### changed in v63.0:
* pref("browser.startup.homepage", "about:home"); // prev: "chrome://branding/locale/browserconfig.properties" `0103`
* pref("security.csp.enable_violation_events", true); // prev: false `2681`

---

### ignore

<details><summary>click me for details</summary><p>

==NEW
```js
pref("accessibility.typeaheadfind.manual", true);
pref("apz.allow_double_tap_zooming", true);
pref("browser.bookmarks.editDialog.showForNewBookmarks", true);
pref("browser.contentblocking.allowlist.annotations.enabled", true);
pref("browser.contentblocking.allowlist.storage.enabled", true);
pref("browser.contentblocking.cookies-site-data.ui.reject-trackers.enabled", true);
pref("browser.contentblocking.cookies-site-data.ui.reject-trackers.recommended", true);
pref("browser.contentblocking.fastblock.control-center.ui.enabled", false);
pref("browser.contentblocking.fastblock.ui.enabled", false);
pref("browser.contentblocking.global-toggle.enabled", false);
pref("browser.contentblocking.introCount", 0);
pref("browser.contentblocking.rejecttrackers.control-center.ui.enabled", true);
pref("browser.contentblocking.rejecttrackers.reportBreakage.enabled", true);
pref("browser.contentblocking.rejecttrackers.ui.enabled", true);
pref("browser.contentblocking.rejecttrackers.ui.recommended", true);
pref("browser.contentblocking.reportBreakage.enabled", false);
pref("browser.contentblocking.reportBreakage.url", "https://tracking-protection-issues.herokuapp.com/new");
pref("browser.contentblocking.trackingprotection.control-center.ui.enabled", true);
pref("browser.contentblocking.trackingprotection.ui.enabled", true);
pref("browser.contentblocking.ui.enabled", true);
pref("browser.ctrlTab.recentlyUsedOrder", true);
pref("browser.fastblock.enabled", false);
pref("browser.fastblock.limit", 0);
pref("browser.fastblock.timeout", 5000);
pref("browser.newtabpage.activity-stream.asrouter.devtoolsEnabled", false);
pref("browser.newtabpage.activity-stream.asrouter.messageProviders", "[{\"id\":\"onboarding\",\"type\":\"local\",\"localProvider\":\"OnboardingMessageProvider\",\"enabled\":false,\"cohort\":0},{\"id\":\"snippets\",\"type\":\"remote\",\"url\":\"https://snippets.cdn.mozilla.net/us-west/bundles/bundle_d6d90fb9098ce8b45e60acf601bcb91b68322309.json\",\"updateCycleInMs\":14400000,\"enabled\":false},{\"id\":\"cfr\",\"type\":\"local\",\"localProvider\":\"CFRMessageProvider\",\"enabled\":false,\"cohort\":\"\"}]");
pref("browser.newtabpage.activity-stream.pocketCta", "{\"cta_button\":\"\",\"cta_text\":\"\",\"cta_url\":\"\",\"use_cta\":false}");
pref("browser.tabs.remote.separatePrivilegedContentProcess", false);
pref("devtools.aboutdebugging.collapsibilities.installedExtension", false);
pref("devtools.aboutdebugging.collapsibilities.otherWorker", false);
pref("devtools.aboutdebugging.collapsibilities.serviceWorker", false);
pref("devtools.aboutdebugging.collapsibilities.sharedWorker", false);
pref("devtools.aboutdebugging.collapsibilities.tab", false);
pref("devtools.aboutdebugging.collapsibilities.temporaryExtension", false);
pref("devtools.aboutdebugging.network-locations", "[]");
pref("devtools.aboutdebugging.new-enabled", false);
pref("devtools.debugger.features.autocomplete-expressions", false);
pref("devtools.debugger.features.component-pane", false);
pref("devtools.debugger.features.map-await-expression", false);
pref("devtools.debugger.features.map-expression-bindings", true);
pref("devtools.inspector.chrome.three-pane-enabled", false);
pref("devtools.promote.accessibility", 1);
pref("devtools.remote.adb.extensionID", "adb@mozilla.org");
pref("devtools.responsive.leftAlignViewport.enabled", false);
pref("devtools.responsive.show-setting-tooltip", true);
pref("dom.animations-api.compositing.enabled", false);
pref("dom.animations-api.getAnimations.enabled", false);
pref("dom.animations-api.implicit-keyframes.enabled", false);
pref("dom.animations-api.timelines.enabled", false);
pref("dom.datatransfer.mozAtAPIs", true);
pref("dom.events.asyncClipboard", true);
pref("dom.events.asyncClipboard.dataTransfer", false);
pref("dom.events.testing.asyncClipboard", false);
pref("dom.ipc.keepProcessesAlive.privileged", 1);
pref("dom.ipc.processCount.privileged", 1);
pref("dom.payments.defaults.saveAddress", true);
pref("dom.payments.defaults.saveCreditCard", false);
pref("dom.payments.response.timeout", 5000);
pref("dom.pointer-lock.enabled", true);
pref("dom.ua_widget.enabled", false);
pref("dom.visualviewport.enabled", false);
pref("dom.vr.process.enabled", false);
pref("dom.vr.service.enabled", false);
pref("dom.webgpu.enable", false);
pref("dom.window.event.enabled", false);
pref("editor.inline_table_editing.enabled_by_default", true);
pref("editor.positioning.enabled_by_default", true);
pref("editor.resizing.enabled_by_default", true);
pref("extensions.getAddons.langpacks.url", "https://services.addons.mozilla.org/api/v3/addons/language-tools/?app=firefox&type=language&appversion=%VERSION%");
pref("extensions.webextensions.enablePerformanceCounters", false);
pref("gfx.webrender.debug.show-overdraw", false);
pref("gfx.webrender.dl.dump-content", false);
pref("gfx.webrender.dl.dump-parent", false);
pref("gfx.webrender.force-disabled", false);
pref("identity.fxaccounts.autoconfig.uri", "");
pref("identity.fxaccounts.commands.enabled", false);
pref("intl.multilingual.enabled", false);
pref("intl.tsf.hack.allow_to_stop_hacking_on_build_17643_or_later", false);
pref("io.activity.enabled", false);
pref("layout.css.motion-path.enabled", false);
pref("layout.css.offset-logical-properties.enabled", false);
pref("layout.css.overflow.moz-scrollbars.enabled", false);
pref("layout.css.overflow-break.intrinsic-size", false);
pref("layout.css.scrollbar-width.enabled", false);
pref("layout.css.unknown-webkit-pseudo-element", false);
pref("layout.css.use-counters.enabled", false);
pref("layout.css.xul-box-display-values.content.enabled", true);
pref("layout.css.xul-tree-pseudos.content.enabled", true);
pref("marionette.log.truncate", true);
pref("media.allowed-to-play.enabled", false);
pref("media.autoplay.allow-extension-background-pages", true);
pref("media.autoplay.allow-muted", true);
pref("media.autoplay.ask-permission", false);
pref("media.autoplay.block-event.enabled", false);
pref("media.autoplay.block-webaudio", false);
pref("media.autoplay.enabled.user-gestures-needed", false);
pref("media.benchmark.vp9.fps", 0);
pref("media.benchmark.vp9.versioncheck", 0);
pref("media.peerconnection.sdp.rust.compare", false);
pref("media.peerconnection.sdp.rust.enabled", false);
pref("media.setsinkid.enabled", false);
pref("network.cookie.quotaPerHost", 150);
pref("network.dns.resolver-thread-extra-idle-time-seconds", 60);
pref("network.http.send_window_size", 1024);
pref("network.proxy.enable_wpad_over_dhcp", true);
pref("network.trr.disable-ECS", true);
pref("network.trr.max-fails", 5);
pref("prio.publicKeyA", "35AC1C7576C7C6EDD7FED6BCFC337B34D48CB4EE45C86BEEFB40BD8875707733");
pref("prio.publicKeyB", "26E6674E65425B823F1F1D5F96E3BB3EF9E406EC7FBA7DEF8B08A35DD135AF50");
pref("privacy.restrict3rdpartystorage.expiration", 2592000);
pref("security.csp.reporting.script-sample.max-length", 40);
pref("security.pki.mitm_detected", false);
pref("services.settings.poll_interval", 86400);
pref("services.sync.prefs.sync.browser.contentblocking.enabled", true);
pref("services.sync.prefs.sync.browser.ctrlTab.recentlyUsedOrder", true);
pref("urlclassifier.trackingAnnotationTable", "test-track-simple,base-track-digest256");
pref("urlclassifier.trackingAnnotationWhitelistTable", "test-trackwhite-simple,mozstd-trackwhite-digest256");
```

==REMOVED or HIDDEN
```js
pref("accessibility.typeaheadfind.enabletimeout", true);
pref("browser.chromeURL", "chrome://browser/content/");
pref("browser.contentHandlers.types.0.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.0.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.0.uri", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.1.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.1.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.1.uri", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.2.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.2.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.2.uri", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.3.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.3.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.3.uri", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.4.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.4.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.4.uri", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.5.title", "chrome://browser-region/locale/region.properties");
pref("browser.contentHandlers.types.5.type", "application/vnd.mozilla.maybe.feed");
pref("browser.contentHandlers.types.5.uri", "chrome://browser-region/locale/region.properties");
pref("browser.newtabpage.activity-stream.asrouter.snippetsUrl", "");
pref("browser.newtabpage.activity-stream.asrouterExperimentEnabled", false);
pref("browser.newtabpage.activity-stream.asrouterOnboardingCohort", 0);
pref("browser.newtabpage.activity-stream.feeds.theme", true);
pref("browser.newtabpage.activity-stream.section.topstories.showDisclaimer", true);
pref("browser.newtabpage.activity-stream.tippyTop.service.endpoint", "https://activity-stream-icons.services.mozilla.com/v1/icons.json.br");
pref("browser.showQuitWarning", false);
pref("devtools.browserconsole.html", false);
pref("devtools.debugger.component-stack-visible", true);
pref("devtools.debugger.features.component-stack", false);
pref("dom.animations-api.element-animate.enabled", true);
pref("extensions.webextensions.themes.enabled", true);
pref("gfx.webrender.async-scene-build", true);
pref("identity.fxaccounts.messages.enabled", false);
pref("layout.accessiblecaret.bar.enabled", false);
pref("layout.accessiblecaret.bar.width", "2.0");
pref("layout.css.all-shorthand.enabled", true);
pref("layout.css.property-append-only", false);
pref("layout.css.unset-value.enabled", true);
pref("media.ogg.flac.enabled", true);
pref("media.wmf.amd.vp9.enabled", false);
pref("network.cookie.prefsMigrated", false);
pref("privacy.trackingprotection.storagerestriction.enabled", false);
pref("services.sync.prefs.sync.browser.ctrlTab.previews", true);
pref("services.sync.prefs.sync.network.cookie.lifetime.days", true);
pref("shield.savant.loglevel", "warn");
pref("svg.marker-improvements.enabled", true);
pref("webgl.webgl2-compat-mode", false);
```

==CHANGED
```js
pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.searchEngines", ""); // prev: "google"
pref("browser.safebrowsing.provider.mozilla.lists", "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,ads-track-digest256,social-track-digest256,analytics-track-digest256,fastblock1-track-digest256,fastblock1-trackwhite-digest256,fastblock2-track-digest256,fastblock2-trackwhite-digest256,fastblock3-track-digest256"); // prev: "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,ads-track-digest256,social-track-digest256,analytics-track-digest256"
pref("browser.translation.engine", "Google"); // prev: "bing"
pref("devtools.accessibility.enabled", true); // prev: false
pref("devtools.debugger.features.skip-pausing", true); // prev: false
pref("devtools.inspector.fonteditor.enabled", true); // prev: false
pref("devtools.inspector.fonthighlighter.enabled", true); // prev: false
pref("devtools.toolbox.selectedTool", "inspector"); // prev: "webconsole"
pref("dom.animations-api.core.enabled", true); // prev: false
pref("dom.clearSiteData.enabled", true); // prev: false
pref("dom.webcomponents.customelements.enabled", true); // prev: false
pref("dom.webcomponents.shadowdom.enabled", true); // prev: false
pref("extensions.webextensions.background-delayed-startup", true); // prev: false
pref("layout.css.prefixes.device-pixel-ratio-webkit", true); // prev: false
pref("layout.css.webkit-appearance.enabled", true); // prev: false
pref("media.av1.enabled", false); // prev: true
pref("network.trr.request-timeout", 1500); // prev: 3000
pref("network.trr.uri", "https://mozilla.cloudflare-dns.com/dns-query"); // prev: ""
pref("urlclassifier.disallow_completions", "test-malware-simple,test-harmful-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,goog-passwordwhite-proto,ads-track-digest256,social-track-digest256,analytics-track-digest256,fastblock1-track-digest256,fastblock1-trackwhite-digest256,fastblock2-track-digest256,fastblock2-trackwhite-digest256,fastblock3-track-digest256"); // prev: "test-malware-simple,test-harmful-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,goog-passwordwhite-proto,ads-track-digest256,social-track-digest256,analytics-track-digest256"
```

</p></details>

**Comment by theWalkingDuck** [comment]
btw how do you grab the prefs ?

**Comment by earthlng** [comment]
with a fixed and slightly edited version of [this](https://github.com/Theemim/GeckoPrefsExporter)

**Comment by Thorin-Oakenpants** [comment]
> deal with Activity-Stream since it's no longer a system addon in FF63

Fun. Follow On Search is also being moved from system addon to internal

**Comment by earthlng** [comment]
63b11 changes since 63b4

#### new
pref("browser.contentblocking.allowlist.annotations.enabled", true);
pref("browser.contentblocking.allowlist.storage.enabled", true);
pref("browser.contentblocking.global-toggle.enabled", true);
pref("browser.contentblocking.reportBreakage.enabled", false); // true in 63b4
pref("browser.fastblock.limit", 0);
pref("browser.newtabpage.activity-stream.asrouter.devtoolsEnabled", false);
pref("browser.newtabpage.activity-stream.asrouter.messageProviders", `string`); // `string` in 63b4
pref("dom.window.event.enabled", false);
pref("editor.inline_table_editing.enabled_by_default", true); // false in 63b4
pref("editor.positioning.enabled_by_default", true); // false in 63b4
pref("editor.resizing.enabled_by_default", true); // false in 63b4
pref("intl.tsf.hack.allow_to_stop_hacking_on_build_17643_or_later", false); // true in 63b4
pref("layout.css.xul-box-display-values.content.enabled", true); // false in 63b4
pref("layout.css.xul-tree-pseudos.content.enabled", true); // false in 63b4
~~pref("services.sync.engine.bookmarks.validation.enabled", true);~~
~~pref("services.sync.engine.passwords.validation.enabled", true);~~


#### removed
pref("browser.newtabpage.activity-stream.asrouterExperimentEnabled", false);
~~pref("geo.provider.ms-windows-location", false);~~

#### changed
~~pref("browser.newtabpage.activity-stream.telemetry.ut.events", true); // prev: false~~
pref("browser.safebrowsing.provider.mozilla.lists", `long-string`); // prev: `long-string`
~~pref("dom.keyboardevent.dispatch_during_composition", true); // prev: false~~
~~pref("font.name-list.sans-serif.zh-TW", `font-list`); // prev: "Arial, PMingLiU, MingLiU, MingLiU-ExtB"~~
~~pref("geo.wifi.uri", "https://location.services.mozilla.com/..."); // prev: "https://www.googleapis.com/..."~~
~~pref("layout.css.moz-document.url-prefix-hack.enabled", false); // prev: true~~
~~pref("network.tcp.tcp_fastopen_enable", true); // prev: false~~
pref("urlclassifier.disallow_completions", `long-string`); // prev: `long-string`

**Comment by Thorin-Oakenpants** [comment]
`browser.contentblocking.*` - We don't need to do anything with these IMO

@fmarier "browser.contentblocking.reportBreakage.enabled" - reporting breakage: is that user initiated (i.e not automatic)? TIA

**Comment by earthlng** [comment]
we should enforce `browser.contentblocking.enabled=true` because cookie blocking doesn't work if it's set to false. I tend to agree that we can ignore the rest of `browser.contentblocking.*`, at least for now.

**Comment by fmarier** [comment]
> "browser.contentblocking.reportBreakage.enabled" - reporting breakage: is that user initiated (i.e not automatic)?

It controls whether or not the "report broken site" button shows up. It doesn't send anything unless the user clicks on it and submits a breakage report.

**Comment by Thorin-Oakenpants** [comment]
Is `browser.contentblocking.ui.enabled` meant to be `false`. [1476217](https://bugzilla.mozilla.org/show_bug.cgi?id=1476217) and gets flipped in 64?

**Comment by earthlng** [comment]
do we want to do something about AS? disable snippets, pocket and telemetry maybe, just in case?
```js
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false);
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
```

They're working on making media autoplay default-prompt but IDK if it will land in FF63 or not.
If it doesn't land I'd say we just keep blocking autoplay for now. Once it lands we can comment it out and use FF's default-prompt.

As for the rest do you see anything we should add? I don't

**Comment by Thorin-Oakenpants** [comment]
FYI: am kinda super uber tied up for a few days. Will wait for earthlng's final diff and will get onto this in about 3 or 4 days. Feel free to move shit to the ignore pile

**Comment by Thorin-Oakenpants** [comment]
new items that pique my interest by name alone (i know some at default false)
- browser.tabs.remote.separatePrivilegedContentProcess
- dom.datatransfer.mozAtAPIs
- dom.webgpu.enable
- holy crap there's a massive spider 2 feet away from me on the wall
- dom.window.event.enabled
- network.http.send_window_size
- network.proxy.enable_wpad_over_dhcp
- privacy.restrict3rdpartystorage.expiration

But yeah, not much here TBH. If you feel like doing the deprecated commit (and perhaps update the sticky issue) that would be cool

**Comment by Thorin-Oakenpants** [comment]
:spider:  update: he's now 4 feet away

**Comment by Thorin-Oakenpants** [comment]
<details><summary>moved from new to ignore</summary><p>

```js
pref("devtools.aboutdebugging.network-locations", "[]");
pref("devtools.aboutdebugging.new-enabled", false);
pref("devtools.aboutdebugging.showSystemAddons", false);
pref("dom.animations-api.compositing.enabled", false);
pref("dom.animations-api.getAnimations.enabled", false);
pref("dom.animations-api.implicit-keyframes.enabled", false);
pref("dom.animations-api.timelines.enabled", false);
pref("dom.vr.process.enabled", false);
pref("dom.vr.service.enabled", false);
pref("extensions.getAddons.langpacks.url", "https://services.addons.mozilla.org/api/v3/addons/language-tools/?app=firefox&type=language&appversion=%VERSION%");
pref("extensions.webextensions.enablePerformanceCounters", false);
pref("identity.fxaccounts.autoconfig.uri", "");
pref("identity.fxaccounts.commands.enabled", false);
pref("layout.css.motion-path.enabled", false);
pref("layout.css.offset-logical-properties.enabled", false);
pref("layout.css.overflow.moz-scrollbars.enabled", false);
pref("layout.css.overflow-break.intrinsic-size", false);
pref("layout.css.scrollbar-width.enabled", false);
pref("layout.css.unknown-webkit-pseudo-element", false);
pref("layout.css.use-counters.enabled", false);
pref("layout.css.xul-box-display-values.content.enabled", true);
pref("layout.css.xul-tree-pseudos.content.enabled", true);
pref("media.benchmark.vp9.fps", 0);
pref("media.benchmark.vp9.versioncheck", 0);
pref("media.peerconnection.sdp.rust.compare", false);
pref("media.peerconnection.sdp.rust.enabled", false);
pref("network.cookie.quotaPerHost", 150);
pref("network.dns.resolver-thread-extra-idle-time-seconds", 60);
pref("security.csp.reporting.script-sample.max-length", 40);
```

</p></details>

**Comment by earthlng** [comment]
```js
/* 0513: disable Follow On Search (FF53+)
 * Just DELETE the XPI file in your System Add-ons directory
 * [1] https://blog.mozilla.org/data/2017/06/05/measuring-search-in-firefox/ ***/
```
follow-on-search is now builtin in FF63 as well ie no longer a system addon. Remove 0513 ? Deleting unwanted SAs is already mentioned in the header anyway.

**Comment by Thorin-Oakenpants** [comment]
Leave it in for ESR60?, just change it to (FF53-62)?

Also, just quietly, `devtools.aboutdebugging.showSystemAddons` (under ignore new) is default false. Wouldn't this be better as true? We could still stick it under system addons, eg 0507 or something - yeah/no?

**Comment by earthlng** [comment]
I think it's behind the telemetry flags now and only works on builtin SE's anyway.

I doubt that too many of our users would ever debug system addons, IDK.

**Comment by Thorin-Oakenpants** [comment]
man I'm having a hard time following your PR

**Comment by earthlng** [comment]
yeah IDK. We can list it at the top "for visibility" so those few who're interested might see it.
System addons become less and less of an issue because FF64 also removed onboarding and Pocket will be builtin. Only 4 left in FF64 atm: 2 webcompat, autofill and screenshots. No idea why the new Firefox Monitor doesn't show up for me, maybe they're not bundling it on purpose for some reason, IDK

**Comment by earthlng** [comment]
63.0 changes since 63b11

#### new
pref("browser.contentblocking.fastblock.control-center.ui.enabled", false); `true in 63b11`
pref("browser.contentblocking.fastblock.ui.enabled", false); `true in 63b11`
pref("browser.contentblocking.global-toggle.enabled", false); `true in 63b11`
pref("browser.contentblocking.rejecttrackers.reportBreakage.enabled", true);
pref("browser.contentblocking.ui.enabled", true); `false in 63b11`
pref("network.trr.max-fails", 5);

#### changed
~~pref("app.update.channel", "beta"); // prev: "release"~~
~~pref("app.update.url.details", "https://www.mozilla.org/%LOCALE%/firefox/beta/notes"); // prev: "https://www.mozilla.org/%LOCALE%/firefox/notes"~~
~~pref("app.update.url.manual", "https://www.mozilla.org/firefox/beta"); // prev: "https://www.mozilla.org/firefox/"~~
~~pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts", true); // prev: false~~
pref("network.trr.request-timeout", 1500); // prev: 3000
pref("network.trr.uri", "https://mozilla.cloudflare-dns.com/dns-query"); // prev: ""
~~pref("toolkit.telemetry.enabled", true); // prev: false~~

**Comment by Thorin-Oakenpants** [comment]
>  because FF64 also removed onboarding and Pocket

Well at least those two have a pref (for now, touch wood). I'm starting to feel like system addon's are test beds (with system addon updates). Maybe it's mandate changed. I thought it's original purpose was for something else. OK, I'll just shove that pref to the list for visibility

**Comment by Thorin-Oakenpants** [comment]
> No idea why the new Firefox Monitor doesn't show up for me

I guess it will turn up in major releases (probably point releases). And I'm on portable which is a complete package each time, and it hasn't shown yet from memory (not sure if in 62.0.3).

I actually thought that it was pushed via system addon updates (outside of full installs) - do you have `user_pref("extensions.systemAddon.update.url` blanked?

**Comment by earthlng** [comment]
> > follow-on-search is now builtin in FF63 as well ie no longer a system addon

> Leave it in for ESR60?, just change it to (FF53-62)?

I'd just remove it. Removing unwanted SAs is mentioned in the section header and the link ... meh.
AFAIK follow-on-search just adds `utm=ff` or something like that to 2ndary etc searches on builtin SEs like google etc. It funds mozilla and is actually a nice way to support them IMO. And it doesn't cost you anything like fe. donations would. The engines already know that someone is using FF anyway.

**Comment by earthlng** [comment]
> do you have user_pref("extensions.systemAddon.update.url blanked?

yes

> Do we need to do anything with F Monitor? Is there a pref, or does it do nothing unless you sign up for it?

IDK I haven't looked at it yet. I don't think you need to sign up for it though

**Comment by Thorin-Oakenpants** [comment]
https://blog.mozilla.org/blog/2018/09/25/introducing-firefox-monitor-helping-people-take-control-after-a-data-breach/

part 2: says to sign up

**Comment by earthlng** [comment]
what about `- TWO: DELETE the XPI file in your System Add-ons directory (note this get reinstalled on app updates)` in 0514? That can go as well IMO

**Comment by Thorin-Oakenpants** [comment]
^^ well we need to deal to AS overall don't we, since it's no longer a SA?

**Comment by earthlng** [comment]
> part 2: says to sign up

well yes if you want to be notified for future breaches but you can also just re-check from time to time without signing up

> part 2: says to sign up

all I was saying is that it doesn't appear to be bundled atm. I've updated my testing nightly and it wasn't even bundled there, that's all I know.

**Comment by earthlng** [comment]
> FF search: do you mean to the deprecated section or entirely?

entirely remove it. Without a pref to it there's no need to move it to 9999 IMO

**Comment by Thorin-Oakenpants** [comment]
^ OK. AFAIK F Monitor wouldn't "do" anything unless it had your details. IDK, maybe it tries to grab it from forms?

If you go to https://github.com/ghacksuserjs/ghacks-user.js - do you see `environments` in the top nav, after commits, branches, releases ? WTF is an environment. It's only just shown up (IDK, in the last week maybe?)

**Comment by earthlng** [comment]
https://blog.mozilla.org/futurereleases/2018/06/25/testing-firefox-monitor-a-new-security-tool/

> In the meantime, check out and download the latest version of Firefox Quantum for the desktop in order to use the Firefox Monitor feature **when it becomes available.**

probably just hasn't been rolled out to everyone yet and therefore not bundled either

**Comment by Thorin-Oakenpants** [comment]
OK, PR looks good (done my local diff) .. just one Q: `do we just remove 3=for n days...`

I assume in FF63+ a value of 3 is invalid? If so then we could perhaps use

current PR
```
...2=until you close Firefox, 3=for n days (see 2704-deprecated)
```

possible
```
2=until you close Firefox, 3=for n days (see 2704-deprecated in FF63+)
```

TBH, I think the ticket said it wasn't even being called? And I would imagine most people use extension cleaners. But the above should do to satisfy ESR? :+1: or nits

**Comment by Thorin-Oakenpants** [comment]
be4 I forget: do we need to do anything with the 2 checkbox options for 3rd party cookies? I haven't got a 63 set up here to check what prefs the options control:
- https://cdn.vox-cdn.com/uploads/chorus_asset/file/13321101/Screen_Shot_2018_10_23_at_3.15.53_PM.png

**Comment by earthlng** [comment]
No. they're controlled by network.cookie.cookieBehavior

**Comment by earthlng** [comment]
```js
/* 0514: disable Activity Stream (FF54+)
 * Activity Stream is the default homepage/newtab in FF57+. It is based on metadata and browsing behavior,
 * and includes telemetry and web content such as snippets, top stories (pocket), top sites, etc.
 *  - ONE: make sure to set your "home" and "newtab" to about:blank (or use an extension to control them)
 *  - TWO: DELETE the XPI file in your System Add-ons directory (note this get reinstalled on app updates)
 * And/or you can try to control the ever-growing, ever-changing "browser.newtabpage.activity-stream.*" prefs
 * [FF63+] Activity Stream (AS) is now builtin and no longer an easily deletable system addon!
 *     We'll clean this up and move to a new number when ESR67 is released.
 * [1] https://wiki.mozilla.org/Firefox/Activity_Stream
 * [2] https://www.ghacks.net/2016/02/15/firefox-mockups-show-activity-stream-new-tab-page-and-share-updates/ ***/
user_pref("browser.library.activity-stream.enabled", false); // (FF57+)
/* 0514a: disable AS Snippets ***/
user_pref("browser.newtabpage.activity-stream.disableSnippets", true);
user_pref("browser.newtabpage.activity-stream.feeds.snippets", false); // [SETTING] Home>Firefox Home Content>Snippets
/* 0514b: disable AS Top Stories and other Pocket-based and/or sponsored content ***/
user_pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
user_pref("browser.newtabpage.activity-stream.section.highlights.includePocket", false); // [SETTING] Home>Firefox Home Content>Highlights>Pages Saved to Pocket
user_pref("browser.newtabpage.activity-stream.showSponsored", false);
/* 0514c: disable AS telemetry ***/
user_pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry", false);
user_pref("browser.newtabpage.activity-stream.telemetry.ping.endpoint", "");
```

**Comment by Thorin-Oakenpants** [comment]
I think we need more than this, but it's a start :+1: just commit it

**Comment by Thorin-Oakenpants** [comment]
no, I was thinking of other prefs that still allow "things to happen" (even if only locally = persistent data) as per one other previous thread which I can't find, but its old now, and I think we should revisit AS in totality at some stage in the near future, and the whole AS morphs too fast to keep track, and at the time we just said "delete the xpi"

Speaking of which, we should drop that line, because as you said, its in the header section. But wait .. there's MOAR. We do not need to keep anything for ESR users. ESR users are on v60, and we have an archived 60 for them. Case in point - you removed the FollowOn Search entry. I don't like the AS section at all now - it's as messy AF. We either do it now (move to proper section), or create a new ticket to do it after beta - please advise

**Comment by Thorin-Oakenpants** [comment]
> please advise

bump

**Comment by earthlng** [comment]
IMO it can wait until ESR67 but if you want to change it now go for it

**Comment by Atavic** [comment]
Some AS entries were discussed on #396

**Comment by Thorin-Oakenpants** [comment]
- Where do we put this? I think it should be under hardware because most of it falls back to that - e.g. battery(it was in the spec)?, resolutions, gpu info(?) etc
- Do we really need the second pref? (I assume the first is a master switch, and even then, the second is false and we could add it if they ever flip it)
- It's all a bit vague on the FP'ing side of things and IANAExpert. I guess since it's only just been enabled, it's early days and we could block it with lilttle to no breakage. [Here](https://bugzilla.mozilla.org/show_bug.cgi?id=1461454) are some RFPs required for Media Capabilities. But overall, from most of my reading, it seems most of this is already available (but is it protected? eg screen res).

```js
/* 2517: disable Media Capabilities API (FF63+)
 [1] https://github.com/WICG/media-capabilities
 [2] https://wicg.github.io/media-capabilities/#security-privacy-considerations ***/
user_pref("media.media-capabilities.enabled", false);
user_pref("media.media-capabilities.screen.enabled", false); // default: false
```

**Comment by earthlng** [comment]
> Should we rename Tracking Protection in the user,js to Content Blocking (or at least append it)?

rename: IMO no because TP is 1 part of Content Blocking. append: sure, why not

> Where do we put this?

yeah under HW makes sense. But I'm not sure if we should add this at all. We keep disabling stuff because it could theoretically be used for FPing while there's no indication that it's really used that way. But disabling it definitely puts us in a very small group. I'm not sure if there's really anything too unique that could be gathered from MC because most devices will have similar capabilities.
The sad thing is that it looks like they have a patch ready to address these issues in MC since **5 months** but it's just sitting there waiting to be looked at and commited :(

> Do we really need the second pref?

no we can ignore the 2nd pref

> we could add it if they ever flip it

that's assuming we keep doing the diffs. This issue is 2 months old and FF63 is out for almost 3 weeks now and we're still working on it. Do we really want to continue doing the ToDo: Diffs issues if we can't finish them in time for a new FF release? Nobody but us 2 contributes and nobody seems to really care anyway.

ps: other things off the top of my head that make us pretty unique for no good reason IMHO ...
- security.tls.version.min = 3
- full-screen-api.enabled = false

**Comment by crssi** [comment]
> nobody seems to really care anyway

I do, and I really appreciate your doing. :heart: 

> Nobody but us 2 contributes

I would, but I have to admit that I am way not as smart as you to :cry: 

I am sure that most of other followers also belong to "my club" regarding upper saying. :wink: ,

**Comment by Thorin-Oakenpants** [comment]
Well, TBH, we (pants and earthlng) do what we do when we do it. The time frame is not set in concrete. The last commit to do with these diffs was 15 days ago (AS stuff), and all that's been left are the two prefs, one of which is at default, and the other we're not even sure on. The master copy is always the up to date version (but I get it, some people might wait for a release).

Yup, it would be nice to always pump these out within a few days, but shit happens (as per my email). I/we could have wrapped this up weeks ago, except I was actually kind of thinking "blah, lets see what happens when *I* (we) do nothing". Seems at times we're not appreciated.

That said, if the information is there (i.e the diffs are generated), then it *will* be actioned - build it and they shall come. This release has just been a bit of a bad time personally. But yeah, I too feel like WTF, why bother anymore.

Anyway, lets wrap this puppy up. I have some questions (see next post).

**Comment by Thorin-Oakenpants** [comment]
`Content Blocking (required to block cookies) (FF63+)` - is this required for ANY control of cookies (and said related localStorage), i.e network.cookie.cookieBehavior could be any value - i.e all cookies now go thru CB?

@fmarier for good measure

**Comment by Thorin-Oakenpants** [comment]
re MC API, I am inclined to disable it for now. Now that it's hit 63, maybe they can get on with those RFP patches (which I am sure will be before the next ESR cycle). It's new (so just imagine you're on ESR and it doesn't exist) so I doubt it will affect end users for quiet a while - we can always revisit it. And we already disable HWA, and DRM/GMP stuff, and I imagine this already affects it? I'd rather be consistent just for now.

**Comment by Thorin-Oakenpants** [comment]
actually, I reopen it - if you're happy then close it, and I'll take that as a signal to create a release

**Comment by earthlng** [comment]
> is this (`browser.contentblocking.enabled`) required for ANY control of cookies

yes. It's the master switch for content blocking which atm is TP and cookie-blocking.
Because we have the cookie part in another section I wouldn't add Content-Blocking to the header or subheader in 0400. This pref was also removed again in FF65 and we can delete it from the user.js for FF65 because nobody needs it anymore then and it's not in ESR either. (or Waterfox, etc)

> That said, if the information is there (i.e the diffs are generated), then it will be actioned

creating the diffs is easy, actioning them is the tedious part but okay I'll create a issue for the current 64 beta diff and we'll see how it goes.

**Comment by Thorin-Oakenpants** [comment]
So why bother to add the pref in the first place. I'll remove it then.

**Comment by Thorin-Oakenpants** [comment]
If you really want it in, then lets add in under 2707 for two releases?

**Comment by earthlng** [comment]
I think more people disable TP than cookie-blocking and that's why it fits better under 0400 IMO.
Let's add it for 2 releases to make sure cookie blocking works as expected.

**Comment by earthlng** [comment]
yes please, it's about time :)

**Comment by Thorin-Oakenpants** [comment]
wot the heck man? I've been waiting on you for ages now :trolleybus:

**Comment by Atavic** [comment]
> nobody seems to really care anyway

I do care and I also appreciate new people coming here, like Mr. bogachenko.
